packages/legend-graph/src/graph-manager/protocol/pure/v1/transformation/pureGraph/to/V1_ElementSecondPassBuilder.ts

```diff
@@ -143,6 +147,30 @@ export class V1_ElementSecondPassBuilder
     );
   }
 
+  visit_SnowflakeApp(element: V1_SnowflakeApp): void {
+    const metamodel = this.context.currentSubGraph.getOwnFunctionActivator(
+      V1_buildFullPath(element.package, element.name),
+    );
+    metamodel.function = PackageableElementExplicitReference.create(
+      guaranteeNonNullable(
+        this.context.graph.functions.find(
+          (fn) =>
+            generateFunctionPrettyName(fn, {
+              fullPath: true,
+              spacing: false,
+            }) === element.function.replaceAll(/\s*/gu, ''),
+        ),
+      ),
+    );
+    metamodel.activationConfiguration =
+      V1_buildSnowflakeAppDeploymentConfiguration(
+        element.activationConfiguration,
+        this.context,
+      );
+
+    observe_SnowflakeApp(metamodel as SnowflakeApp);
```

Comment:
we should never do observe logic in the building phase of packageable elements. this is something you should check before doing initially reviews.